### PR TITLE
(maint) Start using clj-parent

### PR DESCRIPTION
**Pull request by kbarber**
This switches us over to use clj-parent for managed dependencies.

Edit (AJ): And updates all the dependencies we're using in the process.

Signed-off-by: Ken Barber <ken@bob.sh>

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -141,7 +147,7 @@
                                   (require 'schema.core)
                                   (schema.core/set-fn-validation! true))]}
              :ezbake {:dependencies ^:replace [[puppetlabs/puppetdb ~pdb-version]
-                                               [org.clojure/tools.nrepl "0.2.3"]]
+                                               [org.clojure/tools.nrepl nil]]
```
Comment (wkalt) [inline project.clj]: @kbarber as I understand it we don't need the nil here or on line 137.

**Comment by wkalt** [comment]
@kbarber this needs a rebase now too

**Comment by ajroetker** [comment]
@puppetlabs-jenkins, test this please

File: project.clj (hunk 1 above)
Comment (ajroetker) [inline project.clj]: unfortunately we need this, I added the notes/comment from this section of the puppetserver project.clj as well to explain why

**Comment by ajroetker** [comment]
Corresponds with https://github.com/puppetlabs/pe-puppetdb-extensions/pull/253

File: src/puppetlabs/puppetdb/config.clj [hunk 2]
```diff
@@ -202,8 +202,8 @@
   [{db-config :database :or {db-config {}} :as config}]
   (when (str/blank? (:subname db-config))
     (throw+
-     {:type ::cli-error
-      :message
+     {:kind ::cli-error
```
Comment (senior) [inline src/puppetlabs/puppetdb/config.clj]: Are these changes required because of upstream version changes?

File: test/puppetlabs/puppetdb/testutils/services.clj [hunk 3]
```diff
@@ -67,6 +68,7 @@
    #'svcs/puppetdb-service
    #'command-service
    #'status-service
+   #'scheduler-service
```
Comment (senior) [inline test/puppetlabs/puppetdb/testutils/services.clj]: what are we using the scheduler service for?

File: src/puppetlabs/puppetdb/config.clj (hunk 2 above)
Comment (ajroetker) [inline src/puppetlabs/puppetdb/config.clj]: Yep!

File: test/puppetlabs/puppetdb/testutils/services.clj (hunk 3 above)
Comment (ajroetker) [inline test/puppetlabs/puppetdb/testutils/services.clj]: the status service now uses the scheduler service
